### ISSUE TITLE
Map control

### ISSUE THREAD

**Issue by Silver1983** [akarnokd/open-ig#980]
Right now the mouswheel scrolls the map up and down and together with Ctrl it zooms in and out. For two reasons I'd like this be be switched:

1) RTS standard: Every RTS game with a zooming map uses the mousewheel for zooming without any other key beying pressed. So that behaviour is more or less standard and expected by the player.

2) The up and down scrolling by mousewheel isn't really that usefull because you can only do vertical movements. Zooming on the other hand is being used constantly. Therefore the action that is being used more often should get the "easier" command.

3) I only even thought about using the Ctrl key for zooming in and out was because my experience with Adobe products. You can't expect that from everyone :)

**Comment by mansam** [comment]
The way the mouse wheel works is consistent with the way that any window with horizontal and vertical scrollbars works. It change it would be inconsistent.

**Comment by akarnokd** [comment]
The current control scheme matches the controls in a web browser. You use CTRL+Scroll to zoom, Scroll to up-down and SHIFT-Scroll to left-right (at least in Chrome).

Besides, zooming isn't that useful to be the primary scroll event. Panning, however, without moving the mouse itself was more convenient so far.

**Comment by Silver1983** [comment]
You are referring to programs. Of course you're right when you're mentioning a consistent behaviour in comparison to Chrome etc. But isn't OIG a game and therefore should be compared to controll schemes in other games? It gets obvious when you're comparing a ground battle in OIG to other RTS:

- Edge scrolling is missing (will be implemented in the future afaik), 
- click and attack is missing (will be implemented in the future afaik) 
- and the way the map is being zoomed is different as well (but will, as you have said, stay this way). So you get a completely different experience than in any other RTS (I am familiar with).

In addition to that:
A game should be controllable with one imput device (if possible), in this case the mouse. There a exeptions of course: Flight simulators, for example, need to have a big assignment of keys. That way they have a steep learning curve but hey, you wanna fly a plane, don't you :) 

The original IG could be played completely with the mouse. There were additional shortcuts but you didn't have to use them. IMO this is the way to go. As simple as possible but with additional ways of controlling the game. Think about the orignal Command & Conquer or even the latest StarCraft: You can play the game without knowing any shortcut. That way you won't have a convenient way of selecting a group or shortcuts to specific points on the map...but you CAN play it completely with the mouse.

The way it is now you NEED to use the keyboard in OIG to play the game. I know there are scollbars on the galaxy and colony map but using them is like using an office program. You have to detach from the action and think about moving the map. Because of the missing "click to attack" and "move to attack" you need to use the keyboard for giving commands as well. You could use the attack or move button, but that is like playing the first WarCraft game. Those things will be implemented in the future and keyboard commands won't be nessesary any more. But: For controlling the mouse they will be nessesary forever if it stays like it is now. And think about it: 

You use NO key for scrolling up and down but you need a key for scrolling left and right and you need a key for zooming in and out. So moving the map works with and without a key. That is not consistent in a game where both movements are being used equally. It makes sense in Chrome, because the thing you're doing most of the time is scrolling up and down. Left and right isn't really that important when surfing the web.
I don't understand why zooming isn't important for you. It is the fastest way of getting from one point to the other. Zoom out, move the cursor to another spot on the map, zoom in. Done.

Sometimes I am under the impression that you are a programmer thinking with a programmers mind. There are many gamers outside who don't even use a computer for surfing the web any more. They use smartphones. A gamer likes this could only rely on his knowledge of other games. Shouldn't that be sufficient?

I remember our discussion a long time ago about a little program to reassociate jar files with java. You said nobody would need it. My workmates at school and even my pupils wouldn't even know what do do without help like this :)

I hope I could give you some insight in my conception if this issue. In the end it's your interpretation of IG and I respect that.

**Comment by mansam** [comment]
@Silver1983 How much original IG did you actually play? OIG is just as playable with only a mouse. In my personal opinion, the control scheme shouldn't be aimed at confused smartphone gamers, it should be aimed primarily at players of IG, should it not?

**Comment by akarnokd** [comment]
This is an issue I don't want to tackle myself. If somebody wants to implement a different control scheme, PR is welcome; don't forget to allow choosing between the current and the new control scheme.

**Comment by Gosu4095** [comment]
I completely agree with you about the mouse wheel. I don't use this function at all, holding right click/using minimaps to navigate is more than sufficient. 

If I could use the mouse to zoom in/out I suppose that would be an improvement although using right and left click on the zoom buttons for me is totally fine.

Right now using A for attack and M for move in battles, it seems a shame that you can't just right click and the selected unit will attack/move. I would support this addition as well as edge scrolling.

To me these improvements seem extremely logical and "the way it should be".

That said I find the game very comfortable to navigate in its current state so I'm personally not really affected.
